### PR TITLE
More asset proxy gas optimizations [Experiment]

### PR DESCRIPTION
**Pull request by Recmo**
## Description

* Avoids ABI decode by marking function external and not using internal functions calls (which would null the gains from `external`).

*fillOrder ERC20*:
Before: 188468
With #767: 180953 181017
After: 179517

*fillOrder ERC721*:

*batchERC20*:

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/contracts/test/exchange/core.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ chaiSetup.configure();
 const expect = chai.expect;
 const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
 // tslint:disable:no-unnecessary-type-assertion
-describe('Exchange core', () => {
+describe.only('Exchange core', () => {
```
Comment (fabioberger) [inline packages/contracts/test/exchange/core.ts]: Remove `.only`

File: packages/contracts/test/exchange/core.ts [hunk 2]
```diff
@@ -125,12 +125,12 @@ describe('Exchange core', () => {
     afterEach(async () => {
         await blockchainLifecycle.revertAsync();
     });
-    describe('fillOrder', () => {
+    describe.only('fillOrder', () => {
```
Comment (fabioberger) [inline packages/contracts/test/exchange/core.ts]: Remove `.only`

File: packages/contracts/test/exchange/core.ts [hunk 3]
```diff
@@ -125,12 +125,12 @@ describe('Exchange core', () => {
...
         beforeEach(async () => {
             erc20Balances = await erc20Wrapper.getBalancesAsync();
             signedOrder = orderFactory.newSignedOrder();
         });
-        it('should transfer the correct amounts when makerAssetAmount === takerAssetAmount', async () => {
+        it.only('should transfer the correct amounts when makerAssetAmount === takerAssetAmount', async () => {
```
Comment (fabioberger) [inline packages/contracts/test/exchange/core.ts]: Remove `.only`

File: packages/contracts/test/exchange/core.ts [hunk 4]
```diff
@@ -142,7 +142,8 @@ describe('Exchange core', () => {
             expect(takerAssetFilledAmountBefore).to.be.bignumber.equal(0);
 
             const takerAssetFillAmount = signedOrder.takerAssetAmount.div(2);
-            await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress, { takerAssetFillAmount });
+            const res = await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress, { takerAssetFillAmount });
+            console.log(res.gasUsed);
```
Comment (fabioberger) [inline packages/contracts/test/exchange/core.ts]: Remove `console.log`

**Comment by abandeali1** [comment]
This is replaced by #773 . Closing this PR.

**Pull request closed without merging**